### ISSUE TITLE
How does comet perform auto_metric_logging? For example, it automatically logs the loss at every training step.  Does it extract the necessary metrics from the computational graph constructed in pytorch?

### ISSUE THREAD

**Issue by nsriniva03** [comet-ml/issue-tracking#460]
## Before Asking:   
- [ ] I have searched the Issue Tracker.   
- [ ] I have searched the Documentation.    

## What is your question related to?
- [x] Comet Python SDK
- [ ] Comet UI
- [ ] Third Party Integrations (Huggingface, TensorboardX, Pytorch Lightning etc.)  

## Code
```
Please paste a code snippet if your question requires it!   
```

**Comment by dsblank** [comment]
> How does comet perform auto_metric_logging? For example, it automatically logs the loss at every training step. 

For auto logging, we have instrumented the Python framework (like pytorch, tensorflow, etc) So when a metric is computed, we automatically log it.

> Does it extract the necessary metrics from the computational graph constructed in pytorch?

Not from the graph, but from the methods called.

Hope that answers your questions! If not, please re-open.
